torchrl/envs/transforms/transforms.py

```diff
@@ -2150,6 +2163,12 @@ def transform_observation_spec(self, observation_spec: TensorSpec) -> TensorSpec
         return observation_spec
 
     def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
+        if self.as_inverse:
+            return tensordict
+        else:
+            return self.unfolding(tensordict)
+
+    def unfolding(self, tensordict: TensorDictBase) -> TensorDictBase:
         # it is assumed that the last dimension of the tensordict is the time dimension
         if not tensordict.ndim or tensordict.names[-1] != "time":
```

Comment:
Still unsure about this. I know its purpose but not sure if it is really needed. Also if we keep it we probably need a transform that adds the time dimension to a tensordict. Because when using CatFrames as an inverse applied to the replay buffer the data does not have the time dimension. At least for me, it errors out every time.

Reply:
What about a warning that says that we assume the last dim to be "time"? The thing we want to avoid it that time is on another dim and the catframe wrongly assumes that the last dim is time.
Another thing we could do: for now raise a warning, in the future change the behaviour and if the "time" dim is defined then we know which dim must be considered time, and if not we assume it's the last.
TLDR now you can just replace this by a warning